src/client/datascience/jupyter/jupyterNotebook.ts

```diff
@@ -168,6 +169,9 @@ export class JupyterNotebookBase implements INotebook {
     public get disposed() {
         return this._disposed;
     }
+    public get isRestarting() {
+        return this._isRestarting;
```

Comment:
I think we might want to discuss this issue as a team.
Not a fan of this. We have 2 way to determine the status of a kernel now.
This property and the status.

Reply:
I would prefer to have a single way of determining this, i.e. using the existing status.
E.g. when restarting a kernel, manually emit `Restarting` and supress all status emissions until it has restarted or restart has failed at which point we can send the current status of the kernel (which would most likely be in a busted state).

Reply:
@rchiodo @IanMatthewHuff what do you think?
Book a meeting to discuss how we address this?

Reply:
Catching up from my interview day. Is the crux of the issue here the way that we handle restart (by swapping to an existing session) doesn't line up with the actual status of the kernel?

Reply:
Yes, thus inn such cases the status isn't doing what we expect it to.